File: test/fixtures/migration-py-to-ts/src/python/processor.py
"""Dataset processor — DEPRECATED, see TypeScript implementation"""
# @deprecated — migrating to src/typescript/processor.ts
import asyncio
from typing import Any


# LEGACY: synchronous processing pipeline
def process_csv(data: list[dict[str, Any]]) -> list[dict[str, Any]]:
    """Process CSV data rows."""
    results = []
    for row in data:
        processed = {k: str(v).strip() for k, v in row.items()}
        results.append(processed)
    return results


# LEGACY: old validation logic
def validate_dataset(dataset: dict[str, Any]) -> bool:
    """Validate dataset structure."""
    required_fields = ["name", "source_url"]
    return all(field in dataset for field in required_fields)


async def async_process(dataset_id: str, data: list[dict]) -> dict:
    """Async processing wrapper."""
    # TODO migrate: this will be replaced by TypeScript queue-based processing
    await asyncio.sleep(0.1)  # simulate processing
    processed = process_csv(data)
    return {"dataset_id": dataset_id, "rows_processed": len(processed), "status": "complete"}

File: test/fixtures/migration-py-to-ts/src/typescript/processor.ts
// MIGRATED: TypeScript implementation of dataset processor
// Replaces src/python/processor.py
import { z } from 'zod';

const DataRow = z.record(z.string());

export function processCsv(data: z.infer<typeof DataRow>[]): z.infer<typeof DataRow>[] {
  return data.map((row) => {
    const processed: Record<string, string> = {};
    for (const [key, value] of Object.entries(row)) {
      processed[key] = String(value).trim();
    }
    return processed;
  });
}

export function validateDataset(dataset: { name?: string; source_url?: string }): boolean {
  return Boolean(dataset.name && dataset.source_url);
}

export async function asyncProcess(
  datasetId: string,
  data: Record<string, string>[],
): Promise<{ dataset_id: string; rows_processed: number; status: string }> {
  const processed = processCsv(data);
  return { dataset_id: datasetId, rows_processed: processed.length, status: 'complete' };
}
